core/src/banking_stage.rs

```diff
@@ -1002,43 +995,81 @@ impl BankingStage {
             return;
         }
 
-        let forwardable_packets =
-            Self::filter_valid_packets_for_forwarding(buffered_packet_batches.iter());
-        let forwardable_packets_len = forwardable_packets.len();
-        let (_forward_result, sucessful_forwarded_packets_count) = Self::forward_buffered_packets(
-            forward_option,
-            cluster_info,
-            poh_recorder,
-            forwardable_packets,
-            data_budget,
-            banking_stage_stats,
+        let mut forward_packet_batches_by_accounts =
+            ForwardPacketBatchesByAccounts::new_with_default_batch_limits(bank_forks);
+        let forwardable_packets_count = Self::filter_valid_packets_for_forwarding(
+            buffered_packet_batches,
+            &mut forward_packet_batches_by_accounts,
```

Comment:
https://github.com/solana-labs/solana/blob/7dc554bbe398eac0e7e8094ffca48c410df6e6f2/core/src/banking_stage.rs#L520-L521 We also have a forward limit of about 10,000 packets a second, so we could factor that into the bucketing as well.

If the total packets in the buckets exceed this, we can just exit and not process the rest of the buffers

Reply:
This is a good idea. Think can set the forward batches capacity to be inline with this limit, come to same effect of this comment https://github.com/solana-labs/solana/pull/25406#discussion_r892981705.

Reply:
Based on that comment, couldn't it also overshoot the `12k` by a lot if a lot of the transactions are low compute unit transactions?

Reply:
convo moved.